### PR TITLE
ci(add-release-ci-script): add the release ci script to publish maven

### LINKED ISSUE #42: Add the release ci script to automatically release the maven package
Add the ci script to automatically release the maven package. And add the release tag  also

### PR DESCRIPTION
**Pull request by hamzashoukat94**
A ci script is added for the release maven package which takes the release version as input. Create the release tag against that and update the pom file as well. Used the GitHub action to automatically create the changelog and use the input title, as the title of the changelog.

closes #42

### CODE REVIEW

**Review by sufyankhanrao — APPROVED** [review]
LGTM

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        required: true
+        default: "Improving API developer experience"
+
+jobs:
+  publish:
+    name: Creating release version
+    runs-on: ubuntu-latest
+    environment: Production
+    steps:
+      - name: Check out Git repository
```
Comment (sufyankhanrao) [inline .github/workflows/release.yml:22]: This should be rename to `Checking out git repository`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/checkout@v2
+
+      - name: Install Java and Maven
```
Comment (sufyankhanrao) [inline .github/workflows/release.yml:25]: This should be rename to `Installing Java and Maven setup`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+          
+      - name: Update version in pom.xml (Release only)
```
Comment (sufyankhanrao) [inline .github/workflows/release.yml:30]: This should be rename to `Updating version in POM`

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        run: mvn -B versions:set -DnewVersion=${{ github.event.inputs.Version }} -DgenerateBackupPoms=false
+      - name: Create Tag
```
Comment (sufyankhanrao) [inline .github/workflows/release.yml:32]: This should be renamed to `Creating tag`

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        id: tag_version
+        uses: mathieudutour/github-tag-action@v6.0
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          custom_tag: ${{ github.event.inputs.Version }}
+          tag_prefix: ""
+          
+      - name: Create release
```
Comment (sufyankhanrao) [inline .github/workflows/release.yml:40]: This should be renamed to `Releasing to Maven Central`

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        uses: ncipollo/release-action@v1
+        with: 
+          tag: ${{ steps.tag_version.outputs.new_tag }}
+          name: Release ${{ github.event.inputs.Title }}
+          body: ${{ steps.tag_version.outputs.changelog }}
+      - name: Release Maven package
```
Comment (sufyankhanrao) [inline .github/workflows/release.yml:46]: This should be renamed to `Releasing to Maven Central`

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        id: release
+        uses: samuelmeuli/action-maven-publish@v1
+        with:
+          gpg_private_key: ${{ secrets.OSSRH_GPG_SECRET_KEY  }}
+          gpg_passphrase: ${{ secrets.OSSRH_GPG_PASSPHRASE  }}
+          nexus_username: ${{ secrets.OSSRH_USERNAME  }}
+          nexus_password: ${{ secrets.OSSRH_PASSWORD }}
+      - name: Run the commit action
```
Comment (sufyankhanrao) [inline .github/workflows/release.yml:54]: This should be renamed to `Committing the version change`

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        if: steps.release.outputs.exit_code == 0
+        uses: devops-infra/action-commit-push@master
+        with:
+           github_token: "${{ secrets.GITHUB_TOKEN }}"
+           add_timestamp: true
+           commit_message: "ci(release-version): update pom.xml [ci skip]"
+           force: false
+           target_branch: ${{ github.ref }}
+      - name: Push changes
```
Comment (sufyankhanrao) [inline .github/workflows/release.yml:63]: This should be renamed to `Pushing changes`

File: .github/workflows/release.yml (hunk 1 above)
Comment (hamzashoukat94) [reply .github/workflows/release.yml]: Renamed as `Check out git repository`

File: .github/workflows/release.yml (hunk 2 above)
Comment (hamzashoukat94) [reply .github/workflows/release.yml]: Renamed as `Install Java and Maven setup`

File: .github/workflows/release.yml (hunk 3 above)
Comment (hamzashoukat94) [reply .github/workflows/release.yml]: Renamed as `Update version in POM`

File: .github/workflows/release.yml (hunk 4 above)
Comment (hamzashoukat94) [reply .github/workflows/release.yml]: Renamed to `Create Tag`. Add a condition to create this on the successful release

File: .github/workflows/release.yml (hunk 5 above)
Comment (hamzashoukat94) [reply .github/workflows/release.yml]: Renamed as `Create changelog for the release and will only run if the release is successfully done

File: .github/workflows/release.yml (hunk 6 above)
Comment (hamzashoukat94) [reply .github/workflows/release.yml]: Renamed as `Release to Maven Central`

File: .github/workflows/release.yml (hunk 8 above)
Comment (hamzashoukat94) [reply .github/workflows/release.yml]: Renamed as `Push the version change`

**Review by sufyankhanrao — APPROVED** [review]
Good work done.

**Pull request merged**